Framework/Algorithms/src/MagFormFactorCorrection.cpp

```diff
@@ -0,0 +1,112 @@
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidAlgorithms/MagFormFactorCorrection.h"
+#include "MantidKernel/MagneticIon.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+
+using namespace Mantid::PhysicalConstants;
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(MagFormFactorCorrection)
+
+using namespace Kernel;
+using namespace API;
+using namespace PhysicalConstants;
+
+void MagFormFactorCorrection::init() {
+  declareProperty(
+      make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input),
+      "Workspace must have one axes with units of Q");
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "Output workspace name.");
+  declareProperty("IonName", "", "The name of the ion: an element symbol with "
+                                 "a number indicating the valence, e.g. Fe2");
+  declareProperty("FormFactorWorkspace", "",
+                  "If specified the algorithm will create a 1D workspace with "
+                  "the form factor vs Q with a name given by this field.");
+}
+
+void MagFormFactorCorrection::exec() {
+  int64_t iax, numAxes, nQ, iQ;
```

Comment:
Do not declare all your variables at the start of a method (FORTRAN style). Declare them where you use them.